### PR TITLE
Convert Full Dataset: FP Behavior

### PR DESCRIPTION
**Pull request by pauladkisson**
This PR adds support to iterate through all the FP behavior sessions of the dataset and convert with behavior interface.

Missing medpc files are converted via MEDPC_RawFilesbyDate folder to the extent possible.

### CODE REVIEW

File: src/lerner_lab_to_nwb/seiler_2024/medpc.py [hunk 1]
```diff
@@ -2,30 +2,78 @@
...
+def get_MSNs(file_path: str) -> list:
+    """Get the MSNs of all sessions in a MedPC file."""
+    with open(file_path, "r") as f:
+        lines = f.readlines()
+    msns = []
+    for line in lines:
+        if line.startswith("MSN: "):
+            msn = line.split("MSN: ")[1].strip()
+            msns.append(msn)
+    return msns
```
Comment (bendichter) [inline src/lerner_lab_to_nwb/seiler_2024/medpc.py]:
```suggestion
def get_MSNs(file_path: str) -> list:
    """
    Get the MSNs of all sessions in a MedPC file.

    Parameters
    ----------
    file_path : str
        The path to the MedPC file from which to extract MSN values.

    Returns
    -------
    list
        A list of MSN values extracted from the file.
    """
    with open(file_path, "r") as file:
        return [line.split("MSN: ")[1].strip() for line in file.readlines() if line.startswith("MSN: ")]
```

File: src/lerner_lab_to_nwb/seiler_2024/seiler_2024_notes.md
Comment (alessandratrapani) [inline src/lerner_lab_to_nwb/seiler_2024/seiler_2024_notes.md:1]: I could use a little more of context for review faster.

File: src/lerner_lab_to_nwb/seiler_2024/seiler_2024behaviorinterface.py [hunk 2]
```diff
@@ -17,9 +18,13 @@ class Seiler2024BehaviorInterface(BaseDataInterface):
...
-    def __init__(self, file_path: str, start_date: str):
-        start_date = start_date.replace("_", "/")
-        super().__init__(file_path=file_path, start_date=start_date)
+    def __init__(self, file_path: str, session_conditions: dict, start_variable: str, verbose: bool = True):
+        super().__init__(
+            file_path=file_path,
+            session_conditions=session_conditions,
+            start_variable=start_variable,
+            verbose=verbose,
+        )
```
Comment (alessandratrapani) [inline src/lerner_lab_to_nwb/seiler_2024/seiler_2024behaviorinterface.py:28]: Would you mind adding a doc string here to explain the input variables?

File: src/lerner_lab_to_nwb/seiler_2024/seiler_2024_convert_session.py [hunk 3]
```diff
@@ -3,18 +3,23 @@
...
+    start_datetime: datetime,
     subject_id: str,
+    session_conditions: dict,
+    start_variable: str,
     experiment_type: Literal["FP", "Opto"],
     experimental_group: Literal["DPR", "PR", "PS", "RR20"],
     stub_test: bool = False,
+    verbose: bool = True,
 ):
     """Convert a session to NWB."""
```
Comment (alessandratrapani) [inline src/lerner_lab_to_nwb/seiler_2024/seiler_2024_convert_session.py:25]: Would you mind adding a doc string also here to explain the input variables?

**Review by alessandratrapani — COMMENTED** [review]
Overall good! There seem to be a lot of exceptions to handle.

**Review by alessandratrapani — APPROVED** [review]

**Comment by pauladkisson** [comment]
> There seem to be a lot of exceptions to handle.

Yes, see latest email to the lerner lab, which follows up on the missing MSNs, problematic sessions and missing csv sessions. --> Will need to take care of those in a follow-up PR.

**Pull request merged**